packages/circuit-ui/components/ColorInput/ColorInput.tsx

```diff
@@ -106,8 +105,35 @@ export const ColorInput = forwardRef<InputElement, ColorInputProps>(
 
     const hasSuffix = Boolean(suffix);
 
+    const handlePaste: ClipboardEventHandler<InputElement> = (e) => {
+      if (
+        !colorPickerRef.current ||
+        !colorInputRef.current ||
+        disabled ||
+        readOnly
+      ) {
+        return;
+      }
+
+      e.preventDefault();
+
+      const pastedText = e.clipboardData.getData('text/plain').trim();
+
+      if (!pastedText || !/^#[0-9A-F]{6}$/i.test(pastedText)) {
+        return;
+      }
+
+      colorPickerRef.current.value = pastedText;
+      colorInputRef.current.value = pastedText.replace('#', '');
+      colorPickerRef.current.dispatchEvent(
+        new Event('change', { bubbles: true }),
+      );
```

Comment:
Have you tested whether the event is dispatched with the updated value? In the PhoneNumberInput component I had to use a hacky workaround to make this work (https://github.com/sumup-oss/circuit-ui/blob/main/packages/circuit-ui/components/PhoneNumberInput/PhoneNumberInput.tsx#L292-L298).

Reply:
Ah, correct, after some testing and tweaking - fixed.